File: server/history-service/src/main/java/com/chess/historyservice/models/external/GameTurn.java
package com.chess.historyservice.models.external;

import lombok.*;


@Getter
@Setter
@NoArgsConstructor
@Builder
@AllArgsConstructor
public class GameTurn {
    private Position initialPosition;
    private Position destinationPosition;
    private PieceType pieceType;
    private PlayerColor playerColor;
    private int turnNumber;
}

File: app/src/interfaces/GameTurn.ts
import { BoardPosition } from './boardPosition';
import { PieceType } from './piece';
import { PlayerColor } from './player';

export interface GameTurn {
  turnId: number;
  initialPosition: BoardPosition;
  destinationPosition: BoardPosition;
  pieceType: PieceType;
  playerColor: PlayerColor;
  turnNumber: number;
}
